client/client_test.go

```diff
@@ -215,3 +215,37 @@ func TestGetActionTimeout(t *testing.T) {
 		})
 	}
 }
+
+func Test_argListFromMongoDBParams(t *testing.T) {
+	type testParams struct {
+		req      *agentpb.StartActionRequest_PTMongoDBSummaryParams
+		expected []string
+	}
+
+	testCases := []*testParams{
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: "secret"}, []string{"--username", "person", "-psecret", "10.20.30.40:555"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: ""}, []string{"--username", "person", "10.20.30.40:555"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "",
+			Password: "secret"}, []string{"-psecret", "10.20.30.40:555"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 65536, Username: "",
+			Password: "secret"}, []string{"-psecret", "10.20.30.40"}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "", Port: 555, Username: "", Password: "secret"},
+			[]string{"-psecret"}},
+
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "", Port: 0, Username: "", Password: ""}, []string{}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "", Port: 0, Username: "王华", Password: `"`},
+			[]string{"--username", "王华", `-p"`}},
+		{&agentpb.StartActionRequest_PTMongoDBSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: "   "}, []string{"--username", "person", "-p   ", "10.20.30.40:555"}},
```

Comment:
won't it fail calling pt-mongodb-summary?

Reply:
Do we have any specs what params pt-mongodb-summary is to accept and what, to refuse?